### PR TITLE
No results found text is not shown when a user search a non existing currency 

### PR DESCRIPTION
**Pull request by dhairyasenjaliya**
### Details
 - Here we need one condition to check if the searched currency is available or not if available we displayed in list if not matched currency then we need to show a warning message "No results found"
 
 - In order to resolve the issue we are checking the current length of all currency objects and when it returns 0 then shows a message also if  all currency objects are 0 then we are also hiding the header "ALL CURRENCIES"
 

### Fixed Issues

$ https://github.com/Expensify/App/issues/9981

### Tests
1. Click the FAB icon > Send/Request money option
2. Press on the currency icon to navigate to the currency Selection page.
3. Search for any invalid currency, No value should be listed, and "No results found" is shown under the search bar. 
4. Clear the search, and check that all options the visible again.

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The Contributor+ will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps

1. Click the FAB icon > Send/Request money option
2. Press on the currency icon to navigate to the currency Selection page.
3. Search for any invalid currency, No value should be listed, and "No results found" is shown under the search bar. 
4. Clear the search, and check that all options the visible again.

- [x] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/47522946/179890487-db7a80dc-8763-4c44-8d7e-3fe7edf4cd52.mov

#### Mobile Web

https://user-images.githubusercontent.com/47522946/179892819-cb5f5472-1294-4ec2-938b-1cb1488a394e.mp4

#### Desktop

https://user-images.githubusercontent.com/47522946/179890500-3da9f1fa-46d8-4efc-b892-8461721f0155.mov

#### iOS

https://user-images.githubusercontent.com/47522946/179890534-5bdfd3d2-b052-43b3-a671-7772c084b001.mp4

#### Android

https://user-images.githubusercontent.com/47522946/179890709-d28e5e75-88d7-4137-b5df-e8cc1246373a.mp4

I have read the CLA Document and I hereby sign the CLA

### CODE REVIEW

**Comment by dhairyasenjaliya** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by dhairyasenjaliya** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by mountiny** [comment]
Thank you! I havent look at the code yet, Rajat will help out with that, but there was one thing which I had to fix the the PR body, this is what you had:
```
$ https://github.com/Expensify/App/issues/<number-of-the-issue>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ [GH_LINK](https://github.com/Expensify/App/issues/9981)
```

This is a common problem we have and I am on the lookout for it haha. We need the issue to be linked as `$ https://github.com/Expensify/App/issues/<number-of-the-issue>` so we can then correctly run our webhooks logic on this. I have updated the body as required so no worries but I am noting this for your future PRs.

And I hope there s many of them to come :)

**Comment by mountiny** [comment]
@dhairyasenjaliya also I have oticed you commits are not signed. I think you will need to sign them, refer to the 8th point in this [guide](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#begin-coding-your-solution-in-a-pull-request) please.

You might need to make a new branch and force push to this one as the signed commits can get quite annoying!

**Review by parasharrajat — CHANGES_REQUESTED** [review]
1. GPG sign your commits. 
2. Add tests. 
3. Fix QA steps.

```
1. Click the FAB icon > Send/Request money option
2. Press on the currency icon to navigate to the currency Selection page.
3. Search for any invalid currency, No value should be listed, and "No results found" is shown under the search bar. 
4. Clear the search, and check that all options the visible again.
```

File: src/pages/iou/IOUCurrencySelection.js [hunk 1]
```diff
@@ -112,6 +114,7 @@ class IOUCurrencySelection extends Component {
     }
 
     render() {
+        const headerMessage = this.state.currencyData.length === 0 ? this.props.translate('common.noResultsFound') : '';
```
[the remark this answers is not in the document]
Comment (mountiny) [reply src/pages/iou/IOUCurrencySelection.js:117]: +1

File: src/pages/iou/IOUCurrencySelection.js [hunk 2]
```diff
@@ -61,7 +61,9 @@ class IOUCurrencySelection extends Component {
      */
     getSections() {
         const sections = [];
-
+        if (this.state.currencyData.length === 0) {
+            return sections;
+        }
```
[the remark this answers is not in the document]
Comment (dhairyasenjaliya) [reply src/pages/iou/IOUCurrencySelection.js:66]: yeah sure for safety will add a condition

**Comment by dhairyasenjaliya** [comment]
> Thank you! I havent look at the code yet, Rajat will help out with that, but there was one thing which I had to fix the the PR body, this is what you had:
> 
> ```
> $ https://github.com/Expensify/App/issues/<number-of-the-issue>
> 
> Do NOT only link the issue number like this: $ #<number-of-the-issue>
> --->
> $ [GH_LINK](https://github.com/Expensify/App/issues/9981)
> ```
> 
> This is a common problem we have and I am on the lookout for it haha. We need the issue to be linked as `$ https://github.com/Expensify/App/issues/<number-of-the-issue>` so we can then correctly run our webhooks logic on this. I have updated the body as required so no worries but I am noting this for your future PRs.
> 
> And I hope there s many of them to come :)

thank you, sure noted for $ links PR

File: src/pages/iou/IOUCurrencySelection.js [hunk 3]
```diff
@@ -61,7 +61,9 @@ class IOUCurrencySelection extends Component {
      */
     getSections() {
         const sections = [];
-
+        if (this.state.searchValue.trim() && !this.state.currencyData.length) {
+            return [];
+        }
```
Comment (mountiny) [inline src/pages/iou/IOUCurrencySelection.js:66]: I think Rajat also meant to move this above the `const sections = [];`.

**Review by mountiny — COMMENTED** [review]
Thanks for the changes!

File: src/pages/iou/IOUCurrencySelection.js (hunk 3 above)
Comment (dhairyasenjaliya) [reply src/pages/iou/IOUCurrencySelection.js:65]: changes added into latest commit with GPG verified :)

**Comment by dhairyasenjaliya** [comment]
> 1. GPG sign your commits.
> 2. Add tests.
> 3. Fix QA steps.
> 
> ```
> 1. Click the FAB icon > Send/Request money option
> 2. Press on the currency icon to navigate to the currency Selection page.
> 3. Search for any invalid currency, No value should be listed, and "No results found" is shown under the search bar. 
> 4. Clear the search, and check that all options the visible again.
> ```

hello, @parasharrajat have added 1. GPG and also fixed 3.QA steps 
but still figuring about 2. Add tests like what I need to add maybe any document regarding that?

**Comment by dhairyasenjaliya** [comment]
> Your first commit is not signed. I think you might need to create these changes again in a new local branch and then force push to this remote branch just to make sure all is good!

yeah was thinking the same since its fails for very first commit

**Review by parasharrajat — APPROVED** [review]
LGTM. This can't be merged until you fix the QA and test steps. 

cc: @mountiny 

#### PR Reviewer Checklist
- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/docs/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/docs/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/docs/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/docs/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

:ribbon: :eyes: :ribbon:  C+ reviewed

**Review by mountiny — APPROVED** [review]
Great! Thank you both for such a quick turnaround, that is how it should be done 💪 

I have just updated the title so it is clearer and easier to follow for our QA team once in staging.

Additionally, @dhairyasenjaliya our contribution guidelines ask first time contributors to only work on one issue at a time and also wait for regressions from the first PR before they can be hired for the next one. Once you are established and we know we can rely on you, these limits are omitted. I see you have had a good proposal on another issue as well, but I just wanted to note we might need to wait for a few days just to make sure this one goes without regression in staging and then you could continue working on other issues. 

Sorry for any inconvenience, I am very happy with your work here and I appreciate it, but this is just a policy we follow for everyone as it happens people just disappear when they are hired for first issue haha :D

**Comment by dhairyasenjaliya** [comment]
> 

yeah, guidelines make sense sure noted all points :D

**Pull request merged**